File: OOP_HW6/Circle.cpp
//
// Created by User on 2021/6/1.
//

#include "Circle.h"
Circle::Circle(const string& input,const int &radius):Shape(input){
    cout<<"constructor in Circle"<<endl;
    this->radius = radius;
}
void Circle::print(){
    Shape::print();
    cout<<" circle, radius "<<this->radius<<", area "<<get_area()<<endl;
}
double Circle::get_area() {
    return this->radius * this->radius * 3.1415926;
}

File: OOP_HW6/Circle.java
class Circle extends Shape{
    private int radius;
    public Circle(String string, int radius){
        super(string);
        this.radius = radius;
    }
    public void print(){
        super.print();
        System.out.println(" circle, radius "+this.radius+", area "+get_area());
    }
    public double get_area(){
        return this.radius*this.radius*Math.PI;
    }
}
